### PR TITLE
Add pfDataToolbar directive with ngDocs/Example and Unit Tests

### PR DESCRIPTION
**Pull request by jeff-phillips-18**

### CODE REVIEW

File: src/views/toolbar/data-toolbar-directive.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+          return $scope.config.viewsConfig && ($scope.config.viewsConfig.currentView === viewId);
+        };
+
+        $scope.checkViewDisabled = function (view) {
+          return $scope.config.viewsConfig.checkViewDisabled && $scope.config.viewsConfig.checkViewDisabled(view);
+        };
+      },
+      link: function (scope, element, attrs) {
+        scope.$watch('config', function () {
+          if (scope.config && scope.config.viewsConfig && scope.config.viewsConfig.views) {
```
Comment (dtaylor113) [inline src/views/toolbar/data-toolbar-directive.js]: Should this be moved to the controller?

Comment (jeff-phillips-18) [inline src/views/toolbar/data-toolbar-directive.js]: No, watching and observing have to be done post compile (ie in the link function)

Comment (dtaylor113) [inline src/views/toolbar/data-toolbar-directive.js]: Thought we could 'watch' in controller:, and 'observe' in the link:

Comment (jeff-phillips-18) [inline src/views/toolbar/data-toolbar-directive.js]: I've always kept them in the link.

**Comment by dtaylor113** [comment]
I think the pf-selector's "Address" is to low in the box:
apf-toolbar

From PF:
pf-toolbar-orig

**Comment by jeff-phillips-18** [comment]
@dtaylor113 Yeah, this is only in the example though. It's fine when used in an application. I just couldn't get the right demo.css settings to fix it here.

**Comment by dtaylor113** [comment]
Weird list item stacking when selecting between "Birth Month" and "Name".  The input box to the right of the pf-selector momentarily appears vertically stacked before collapsing into the single placeholder.  Is this just in the demo?

**Comment by jeff-phillips-18** [comment]
Yes, again it's in the example only.

File: src/views/toolbar/data-toolbar-directive.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      $scope.listConfig = {
+        selectionMatchProp: 'name',
+        checkDisabled: false
+      };
+    }
+  ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.views').directive('pfDataToolbar',
```
Comment (dtaylor113) [inline src/views/toolbar/data-toolbar-directive.js]: I don't think a 'toolbar' should be under 'views', I know it consumes the 'views' mechanism, but it should probably go into a general 'components' directive.  For now it's fine.

**Comment by dtaylor113** [comment]
LGTM

**Comment by jeff-phillips-18** [comment]
@waldenraines Do you have a moment to review this?

**Comment by waldenraines** [comment]
@jeff-phillips-18 I do but too late :frowning:

**Pull request merged**